### PR TITLE
feat(behavior_path_planner): account for delay before departure in predicted path

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
### <samp>🤖 Generated by Copilot at a8f538b</samp>

Added delayed departure feature for dynamic objects in `behavior_path_planner`. This allows the planner to account for the time that a vehicle might wait before moving from a stop or yield sign. The feature is implemented in `objects_filtering.cpp` and documented in `objects_filtering.hpp`.

should be merged after this [PR](https://github.com/autowarefoundation/autoware.universe/pull/5014)

## Tests performed

- [x] [PASS TIERIV INTERNAL SCENARIOS ](https://evaluation.tier4.jp/evaluation/reports/b144c585-c14e-5970-8e53-23abc5219d98?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utils/path_safety_checker/objects_filtering.cpp [hunk 1]
```diff
@@ -240,15 +239,28 @@ std::vector<PoseWithVelocityStamped> createPredictedPath(
...
 
   std::vector<PoseWithVelocityStamped> predicted_path;
   const auto vehicle_pose_frenet =
     convertToFrenetPoint(path_points, vehicle_pose.position, ego_seg_idx);
 
-  for (double t = 0.0; t < time_horizon + 1e-3; t += time_resolution) {
-    const double velocity =
-      std::clamp(current_velocity + acceleration * t, min_velocity, max_velocity);
-    const double length = current_velocity * t + 0.5 * acceleration * t * t;
+  for (double t = 0.0; t < delay_until_departure + time_horizon + 1e-3; t += time_resolution) {
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/src/utils/path_safety_checker/objects_filtering.cpp:248]: This `delay_until_departure` term is necessary?

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/src/utils/path_safety_checker/objects_filtering.cpp:248]: @satoshi-ota 
Thanks!
You are right, this `delay_until_departure` is not necessary.

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/src/utils/path_safety_checker/objects_filtering.cpp]: @satoshi-ota 
changed in efee117ed8a6f8110af4057019d742cc9e9d2890

**Review by satoshi-ota — APPROVED** [review]
LGTM!

**Pull request merged**